app/routes/event/Events.ts

```diff
@@ -0,0 +1,158 @@
+import { INTERNAL_SERVER_ERROR, OK, NOT_FOUND } from 'http-status-codes';
+import { connection, STATES, Types } from 'mongoose';
+import * as Router from 'koa-router';
+import { setResponse } from '../utils';
+
+import {
+    IApiResponse,
+    IEventModel,
+    TaskModel,
+    SessionModel,
+    IUserSession,
+    AssignmentModel,
+    AssignmentStatus,
+    ICourseStudent,
+    CourseStudentModel,
+} from '../../models';
+
+export const createPostEventsRoute = async (ctx: Router.IRouterContext) => {
+    try {
+        const userSession: IUserSession = ctx.state.user!;
+        const body = ctx.request.body;
+        const { _id: userId } = userSession;
+        const { type, courseId } = body;
+        switch (type) {
+            case 'task': {
+                const event = new TaskModel({
+                    ...body,
+                    author: userId,
+                });
+                ctx.body = {};
+                ctx.body = await event.save();
+                ctx.status = OK;
+                await createAssignments(event, courseId);
+                break;
+            }
+            case 'session': {
+                const event = new SessionModel(body);
+                ctx.body = await event.save();
+                ctx.status = OK;
+                break;
+            }
+            default:
+                return;
+        }
+    } catch (e) {
+        ctx.status = INTERNAL_SERVER_ERROR;
+        ctx.logger.error(e, 'Failed to save document');
+    }
+};
+
+const createAssignments = async (event: IEventModel, courseId: string) => {
+    const students: ICourseStudent[] = await getStudentsByCourseId(courseId);
+    const assignments: any = students.map((student: ICourseStudent) => {
+        const assignment = new AssignmentModel({
+            courseId,
+            deadlineDate: event.endDateTime,
+            mentorId: student.mentors,
+            score: 0,
+            status: AssignmentStatus.Assigned,
+            studentId: student.userId,
+            taskId: event.id,
+        });
+        return { insertOne: { document: assignment } };
+    });
+    await AssignmentModel.bulkWrite(assignments);
+};
+
+export const createDeleteEventsRoute = async (ctx: Router.IRouterContext) => {
+    const { id, eventType } = ctx.params;
+    try {
+        const query =
+            eventType === 'session' ? await SessionModel.findByIdAndRemove(id) : await TaskModel.findByIdAndRemove(id);
+        if (query !== null) {
+            await AssignmentModel.remove({ taskId: id });
```

Comment:
Do you try to remove assignment in case deleting session?

Reply:
Yes, fixed.